client/src/providers/hyperbridge.rs

```diff
@@ -0,0 +1,156 @@
+use crate::types::{
+    filter_map_system_events, system_events_key, BoxStream, Extrinsic, HyperBridgeConfig,
+    LeafIndexQuery,
+};
+use codec::{Decode, Encode};
+use ethers::prelude::H256;
+use hex_literal::hex;
+use ismp::{
+    consensus::{StateCommitment, StateMachineHeight, StateMachineId},
+    events::StateMachineUpdated,
+    host::StateMachine,
+    messaging::{Message, Proof},
+    router::Request,
+};
+use sp_core::{blake2_128, storage::StorageChangeSet};
+use subxt::{rpc_params, OnlineClient};
+
+#[derive(Debug, Clone)]
+pub struct HyperBridgeClient {
+    // WS RPC url of a hyperbridge node
+    pub rpc_url: String,
+    // An instance of Hyper bridge client using the default config
+    pub client: OnlineClient<HyperBridgeConfig>,
+}
+
+impl HyperBridgeClient {
+    pub async fn new(rpc_url: String) -> Result<Self, anyhow::Error> {
+        let client = OnlineClient::<HyperBridgeConfig>::from_url(rpc_url.clone()).await?;
+
+        Ok(Self { rpc_url, client })
+    }
+
+    pub async fn get_current_timestamp(&self) -> Result<u64, anyhow::Error> {
+        let timestamp_addr: [u8; 32] =
+            hex!("f0c365c3cf59d671eb72da0e7a4113c49f1f0515f462cdcf84e0f1d6045dfcbb");
+        let timestamp_op = self.client.rpc().storage(&timestamp_addr, None).await?;
+
+        return if let Some(timestamp) = timestamp_op {
+            let hyper_bridge_timestamp: u64 = codec::Decode::decode(&mut &*timestamp.0).unwrap();
+            Ok(hyper_bridge_timestamp)
+        } else {
+            Ok(0u64)
+        };
+    }
+
+    pub async fn query_request(
+        &self,
+        source_chain: &StateMachine,
+        dest_chain: &StateMachine,
+        nonce: u64,
+    ) -> Result<Vec<Request>, anyhow::Error> {
+        let build_leaf_index_query =
+            LeafIndexQuery { source_chain: *source_chain, dest_chain: *dest_chain, nonce };
```

Comment:
The only field in LeafIndexQuery is just the request or response commitment, pull changes from main